doc/introduction/mid_circuit_measurements.rst

```diff
@@ -0,0 +1,452 @@
+and the :doc:`how-to on creating dynamic circuits with mid-circuit measurements
+<demos/tutorial_how_to_create_dynamic_mcm_circuits>`.
+
+.. _mid_circuit_measurements_statistics:
+
+Mid-circuit measurement statistics
+----------------------------------
+
+Statistics of mid-circuit measurements can be collected along with terminal measurement statistics.
+Currently, :func:`~.counts`, :func:`~.expval`, :func:`~.probs`, :func:`~.sample`, and :func:`~.var`
+are supported.
+
+.. code-block:: python3
+
+    dev = qml.device("default.qubit", wires=2)
+
+    @qml.qnode(dev)
+    def func(x, y):
+        qml.RX(x, wires=0)
+        m_0 = qml.measure(0)
+        qml.cond(m_0, qml.RY)(y, wires=1)
+        return qml.probs(wires=1), qml.probs(op=m_0)
+
+Executing this ``QNode``:
+
+.. code-block:: pycon
+
+    >>> func(np.pi / 2, np.pi / 4)
+    (tensor([0.9267767, 0.0732233], requires_grad=True),
+     tensor([0.5, 0.5], requires_grad=True))
+
+Users can also collect statistics on mid-circuit measurements manipulated using arithmetic/boolean operators.
+This works for both unary and binary operators. To see a full list of supported operators, refer to the
+:func:`~.pennylane.measure` documentation. An example for collecting such statistics is shown below:
+
+.. code-block:: python3
+
+    import pennylane as qml
+
+    dev = qml.device("default.qubit")
+
+    @qml.qnode(dev)
+    def circuit(phi, theta):
+        qml.RX(phi, wires=0)
+        m_0 = qml.measure(wires=0)
+        qml.RY(theta, wires=1)
+        m_1 = qml.measure(wires=1)
+        return qml.sample(~m_0 - 2 * m_1)
+
+Executing this ``QNode``:
+
+.. code-block:: pycon
+
+    >>> circuit(1.23, 4.56, shots=5)
+    array([-1, -2,  1, -1,  1])
+
+Collecting statistics for mid-circuit measurements manipulated using arithmetic/boolean operators is supported
+with :func:`~.counts`, :func:`~.expval`, :func:`~.sample`, and :func:`~.var`.
+
+Moreover, statistics for multiple mid-circuit measurements can be collected by passing lists of mid-circuit
+measurement values to the measurement process:
+
+.. code-block:: python3
+
+    import pennylane as qml
+
+    dev = qml.device("default.qubit")
+
+    @qml.qnode(dev)
+    def circuit(phi, theta):
+        qml.RX(phi, wires=0)
+        m_0 = qml.measure(wires=0)
+        qml.RY(theta, wires=1)
+        m_1 = qml.measure(wires=1)
+        return qml.sample([m_0, m_1])
+
+Executing this ``QNode``:
+
+.. code-block:: pycon
+
+    >>> circuit(1.23, 4.56, shots=5)
+    array([[0, 1],
+           [1, 1],
+           [0, 1],
+           [0, 0],
+           [1, 1]])
+
+Collecting statistics for sequences of mid-circuit measurements is supported with
+:func:`~.counts`, :func:`~.probs`, and :func:`~.sample`.
+
+.. warning::
+
+    When collecting statistics for a sequence of mid-circuit measurements, the
+    sequence may not contain arithmetic expressions.
+
+
+Simulation of mid-circuit measurements
+======================================
```

Comment:
I think the use of `==` makes this a page heading, not a section heading, hence why the rendered docs look the way they currently do. You might need to replace the `==` with `**`. And you may have to add a separate heading for the section containing info about available MCM features.

Reply:
Ahhh, yeah, that makes sense! Thanks for spotting this!

Reply:
Also important because it's causing this to appear as a separate quickstart in the docs.